src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2398,19 +2390,7 @@ def derive_scrypt(self, key_material, salt, length, n, r, p):
             length,
         )
         if res != 1:
-            errors = self._consume_errors()
```

Comment:
This is a bit more debatable. We will error with MemoryError no matter what so it won't proceed unsafely, but it is hypothetically possible this wouldn't be due to OOM. We could hedge by attaching the error stack here like we do in DH

Reply:
Attaching the error stack feels safe in basically all cases.